core/backend/src/IModelDb.ts

```diff
@@ -169,6 +169,13 @@ const withBriefcaseDb = async (briefcase: OpenBriefcaseArgs, fn: (_db: Briefcase
   }
 };
 
+function isDataTransformRequiredError(error: unknown): error is ITwinError {
```

Comment:
@khanaffan @ColinKerr 
This feels messy. I cannot use ITwinError typesafe as shown below. We rethrow a new error sometimes which only preserves the "errorNumber" field which was accessed on :any.
Using ITwinError.isError with a string seems like a better solution, or maybe having an overload that takes an error number...
I would prefer some standard way for doing this.

Reply:
We have inconsistencies across the stack, we need to document best practices and code so we are consistent.  Created a new issue to track larger issue and linked in one other existing issue on same topic.  #8774

@rschili  please add this inconsistency to the larger issue and link it here before you resolve this.